internal/pkg/levee/propagation/propagation.go

```diff
@@ -151,14 +151,14 @@ func (prop *Propagation) visit(n ssa.Node, maxInstrReached map[*ssa.BasicBlock]i
 
 		// This is to avoid attaching calls where the source is the receiver, ex:
 		// core.Sinkf("Source id: %v", wrapper.Source.GetID())
-		if recv := t.Call.Signature().Recv(); recv != nil && source.IsSourceType(prop.config, prop.taggedFields, recv.Type()) {
+		if recv := t.Call.Signature().Recv(); recv != nil && sourceResult.IsSourceType(prop.config, prop.taggedFields, recv.Type()) {
```

Comment:
This seems to be the only place where this function is used. I don't think this function should be exposed by the `source` analyzer. I think that function should be owned by a different package, which the `source` analyzer and other analyzers would use. (e.g. see my comment on #239).

As it is, having this function in the source package, and especially on its `ResultType`, feels confusing to me. Why would a `map` from functions to source nodes know how to identify source types? Why does it need to be fed the config and tagged fields? I think this should be an `IsSourceType` function on a `Classifier` that the `Propagation` would hold a reference to. This would simplify this and other calls.